lib/frontend/delete-github-subscription.js

```diff
@@ -48,24 +64,32 @@ module.exports = async (req, res) => {
       const role = await getRole({ login, installation: userInstallation });
       if (role !== 'admin') {
         res.status(401);
-        return res.json({
+        res.json({
           err: `Failed to delete subscription for ${req.body.installationId}. User does not have access to that installation.`,
         });
+        return;
       }
 
-      await Subscription.uninstall({
-        installationId: req.body.installationId,
-        host: req.body.jiraHost,
-      });
-      return res.sendStatus(202);
+      /** @type {import('../models/subscription')} */
+      const subscription = await Subscription.getSingleInstallation(req.body.jiraHost, req.body.installationId);
```

Comment:
I dont get why we deleted uninstall

Reply:
We move it down to line 81 where we call `destroy`. `uninstall()` does a `get` and then a `destroy` but we need to know the subscriptions details to record it before we destroy it.

Reply:
Would it make sense to make uninstall and install do the logging so we don't have to duplicate uninstall logic?

Reply:
Hmm, maybe, the reason I didn't was because install/uninstall don't know the source of the request, lemme look at the possible collateral damage of adding a parameter to install/uninstall.

Reply:
`Subscription` models don't have any explicit linkage to `Installation` models (which feels like a design flaw, but it's what we have right now). I can link the two up together, but it means importing the `Installation` model in or having an optional `installationId` parameter for uninstalling to be able to report the install ID. I think the method I've implemented here keeps the Subscription from needing knowledge about Installations. I looked into implementing `hasMany` and `belongsTo` but because there are no uniqueness guarantees in the database for anything but primary key, this process also breaks down.

Summary:
 I think that this process outlined here is the safest method without a larger refactor to improve the data model/data integrity.